packages/common/workspace/src/engine/blob.ts

```
@@ -122,6 +145,19 @@ export class BlobEngine {
       throw new Error('local peer is readonly');
     }
 
+    let shouldProceed = true;
+
+    const updateShouldProceed = (proceed: boolean) => {
+      shouldProceed = proceed;
+    };
+
+    this.onBlobSet.emit({ key, value, updateShouldProceed });
+
+    if (!shouldProceed) {
+      logger.error('blob over limit, abort set');
+      return key;
```

should we throw a error, and maybe blocksuite will show error view?

---

What do you think about this? @regischen

---

I feel that it mainly depends on the product side.

---

Now, if a blob is not found, there will be a corresponding style. I feel that it might also be necessary to stop creating blocks after exceeding the limit.

---

Feels that a reminder should suffice, there's no need to prevent the creation of blocks.